### PR TITLE
Add runGSEA.py script

### PR DESCRIPTION
**Pull request by reshu23**
Don't merge yet. Documentation will be added.

### CODE REVIEW

**Comment by sebastian-luna-valero** [comment]
Thanks @reshu23 

Please let me know when the scripts is ready for review.

**Comment by reshu23** [comment]
@sebastian-luna-valero : The script is ready for review. I have implemented testing. All fuctionality is working. Would you be able merge it in to master,If there are no issues? Thank you

**Comment by sebastian-luna-valero** [comment]
Many thanks @reshu23, will have a look

**Comment by sebastian-luna-valero** [comment]
Also, I was wondering whether @haniap could have a look as well before merging.

File: CGAT/scripts/runGSEA.py [hunk 1]
```diff
@@ -0,0 +1,1619 @@
...
+Mootha, Lindgren, et al. (2003, Nat Genet 34, 267-273).
+
+=============================================
+coding: utf-8
+--------------------
+Command line options
+--------------------
+'''
+import sys
+import CGAT.Experiment as EW
```
Comment (sebastian-luna-valero) [inline CGAT/scripts/runGSEA.py]: Please use `import CGAT.Experiment as E` to be consistent with the rest of CGAT scripts

File: CGAT/scripts/runGSEA.py [hunk 2]
```diff
@@ -0,0 +1,1619 @@
...
+import matplotlib.pyplot as plt
+import matplotlib.font_manager as font_manager
+import matplotlib.lines as mlines
+import csv
+import random
+import statsmodels.sandbox.stats.multicomp as sm
+from matplotlib.colors import ListedColormap
+from decimal import *
+import pandas as pd
+import random as RND
```
Comment (sebastian-luna-valero) [inline CGAT/scripts/runGSEA.py]: I would just use `import random` instead to make the code more readable afterwards.

File: CGAT/scripts/runGSEA.py [hunk 3]
```diff
@@ -0,0 +1,1619 @@
...
+import string
+import itertools as ITL
```
Comment (sebastian-luna-valero) [inline CGAT/scripts/runGSEA.py]: I would just use `import itertools` instead to make the code more readable afterwards.

File: CGAT/scripts/runGSEA.py [hunk 4]
```diff
@@ -0,0 +1,1619 @@
...
+    # dict_new is dictionary format of expression dataset "ind_dict"
+    # gene_set_a is stored included gene set "GG"
+    # id_for_leading is id of expression dataset. I am using array format "id_new"
+    # Apart from that, I am using store_gene_leading_info as SGLI, original_es_index as OEI, original_es as OE
+    # I am also using up_for_plot and down_for_plot as ufp and dfp, which stores information for retrieving
+    # original index in original_nes.
+    # ch is choice of user FDR and hm is selected number of genes for leading edge analysis
+    # matr is temp matrix, used by me earlier.
+    # This section is little bit complex.
+    #######################################################################
```
Comment (sebastian-luna-valero) [inline CGAT/scripts/runGSEA.py]: Should this be a `python docstring` instead?

For reference:
* http://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html
* https://www.python.org/dev/peps/pep-0257/

Comment (sebastian-luna-valero) [inline CGAT/scripts/runGSEA.py]: The same could be applied to other functions as well.

File: CGAT/scripts/runGSEA.py [hunk 5]
```diff
@@ -0,0 +1,1619 @@
...
+        "upregulated")
+    plot_summary_report(
+        nes_down_index,
+        original_nes,
+        down_for_plot,
+        in_list,
+        "downregulated")
+
+    # Generate pvalue vs ES graph.
+    AW = down_for_plot[nes_down_index]
```
Comment (sebastian-luna-valero) [inline CGAT/scripts/runGSEA.py]: I would use lowercase for variables and give them meaningful names so it's easier to follow the code.

**Comment by Acribbs** [comment]
I have had a look at the script and tested its output, it all looks ok from my end and should be merged if no-one has any other issues.

**Pull request merged**